src/query-builder/transformer/RawSqlResultsToEntityTransformer.ts

```diff
@@ -46,21 +61,39 @@ export class RawSqlResultsToEntityTransformer {
     transform(rawResults: any[], alias: Alias): any[] {
         const group = this.group(rawResults, alias)
         const entities: any[] = []
-        group.forEach((results) => {
+        for (const results of group.values()) {
             const entity = this.transformRawResultsGroup(results, alias)
-            if (
-                entity !== undefined &&
-                !Object.values(entity).every((value) => value === null)
```

Comment:
This check is removed because it is expensive and is unnecessary. It was originally added in #7450 along with accompanying tests, and the tests continue to pass even after removing it. Additional checks have been added to the code since then which keep track of entities where all values are null (via `hasData` variables).